packages/devtools_app/lib/src/memory/memory_diffing.dart

```diff
@@ -0,0 +1,504 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:devtools_app/src/core/message_bus.dart';
+//import 'package:devtools_app/src/theme.dart';
+import 'package:devtools_app/src/ui/icons.dart';
+import 'package:flutter/material.dart';
+
+import '../shared/globals.dart';
+import 'memory_controller.dart';
+import 'memory_graph_model.dart';
+import 'memory_snapshot_models.dart';
+
+// Diff Image.
+/*
+Image diffImage(BuildContext context) {
+  final themeData = Theme.of(context);
+  return createImageIcon(
+    themeData.isDarkTheme
+        ? 'icons/memory/communities_white.png'
+        : 'icons/memory/communities_black.png',
+  );
+}
+*/
+
+ThemedImageIcon diffImage(BuildContext context) {
+  // TODO(terry): Match shape in event pane.
+  return const ThemedImageIcon(
+    darkModeAsset: 'icons/memory/communities_white.png',
+    lightModeAsset: 'icons/memory/communities_black.png',
+  );
+}
+
+class SimpleObject {
+  SimpleObject(this.id);
+
+  int id;
+}
+
+class ListComputation<T> {
+  ListComputation(List<T> previousList, List<T> currentList, this.valueFunction)
+      : deletedOnes = previousList.toList(),
+        newOnes = currentList.toList();
+
+  /// Items newly created (did not exist in previous).
+  final List<T> newOnes;
```

Comment:
Add a TODO to use sets instead of List. Right now I believe this logic is O(N^2) number of objects in the new and old lists but it can be O(N) if we leverage using sets or maps where the identity hashcode is the key. 
Gotcha is there might be a couple identity hash collisions we need to handle gracefully.